csds-client/client/util/util.go

```diff
@@ -392,3 +346,98 @@ func PrintDetailedConfig(response proto.Message, opts ClientOptions) error {
 	}
 	return nil
 }
+
+// ConnWithJwt connects to uri with jwt authentication
+func ConnWithJwt(opts client.ClientOptions) (*grpc.ClientConn, error) {
```

Comment:
Passing opts makes interface rather obscure. There is no way to tell by the caller of the function what is really needed in order to connect with JWT. 

I would call this function ConnToGCPWithJwt(jwt Jwt, uri string) and make it very specific for GCP. Since we do not support any other platform - generalizing  to support multiple platforms seems like a premature optimization now.